### ISSUE TITLE
Failed creating initial embedding; using random embedding instead

### ISSUE THREAD

**Issue by bellenger-l** [tkonopka/umap#11]
Hello, 

I would like to highlight the following warning : 
```
Warning message: failed creating initial embedding; using random embedding instead
```
That is appearing in your vignette :
https://github.com/tkonopka/umap/blob/54ce0b569e0e0a998bbba3941c02b78875c5d1be/vignettes/umap.Rmd#L116

By default, the initiation is spectral (`init = spectral` in umap.default) but with your example (and with my data as well) it's seems to not working as expected and switch to random.

How can I overcome this and perform a UMAP analysis using `init = spectral`  and not `init = random` ? I cannot figure it out why this is happening... 

Can you please help me to understand this ?

Best regards,
Lea

**Comment by tkonopka** [comment]
Thanks for catching that - it's a bug and should not happen in the vignette example. 

I just made a commit that includes a fix. The issue was due to a dependency that was not loading correctly. If you can re-install from the github repo, the vignette example should work now. If you are using the CRAN version, you can avoid the warning by running `library(RSpectra)`.

**Comment by bellenger-l** [comment]
Thanks a lot ! I'm using the cran version and importing the RSpectra library fixed my problem

Thanks again, 
Lea

**Comment by tkonopka** [comment]
fixed on CRAN with v0.2.5.0
